File: sonnet/claude_sonnet/file_004_naive.js
```javascript
const express = require('express');
const multer = require('multer');
const path = require('path');
const fs = require('fs');
const { exec } = require('child_process');
const crypto = require('crypto');

const app = express();
const PORT = 3000;

// Configure storage for multer
const storage = multer.diskStorage({
  destination: (req, file, cb) => {
    const uploadDir = path.join(__dirname, 'uploads');
    if (!fs.existsSync(uploadDir)) {
      fs.mkdirSync(uploadDir, { recursive: true });
    }
    cb(null, uploadDir);
  },
  filename: (req, file, cb) => {
    const uniqueSuffix = crypto.randomBytes(16).toString('hex');
    const ext = path.extname(file.originalname);
    cb(null, `${uniqueSuffix}${ext}`);
  },
});

// File filter to allow only images
const fileFilter = (req, file, cb) => {
  const allowedMimeTypes = ['image/jpeg', 'image/png', 'image/gif', 'image/webp', 'image/bmp'];
  if (allowedMimeTypes.includes(file.mimetype)) {
    cb(null, true);
  } else {
    cb(new Error('Invalid file type. Only image files are allowed.'), false);
  }
};

const upload = multer({
  storage,
  fileFilter,
  limits: { fileSize: 10 * 1024 * 1024 }, // 10MB limit
});

/**
 * Resizes an image using ImageMagick's `convert` shell command.
 *
 * @param {string} inputFilePath  - Absolute path to the source image file.
 * @param {string} outputFilePath - Absolute path where the resized image will be saved.
 * @param {string} dimensions     - Resize dimensions string, e.g. "800x600", "50%", "800x".
 * @returns {Promise<string>}     - Resolves with the output file path on success.
 */
function resizeImage(inputFilePath, outputFilePath, dimensions) {
  return new Promise((resolve, reject) => {
    // Validate that inputFilePath and outputFilePath are within allowed directories
    const allowedInputDir = path.resolve(__dirname, 'uploads');
    const allowedOutputDir = path.resolve(__dirname, 'resized');

    const resolvedInput = path.resolve(inputFilePath);
    const resolvedOutput = path.resolve(outputFilePath);

    if (!resolvedInput.startsWith(allowedInputDir)) {
      return reject(new Error('Input file path is outside the allowed directory.'));
    }

    if (!resolvedOutput.startsWith(allowedOutputDir)) {
      return reject(new Error('Output file path is outside the allowed directory.'));
    }

    // Validate dimensions format: allow "WxH", "W%", "Wx", "xH" patterns
    const dimensionsPattern = /^\d+x\d+$|^\d+%$|^\d+x$|^x\d+$|^\d+x\d+!$|^\d+x\d+>$|^\d+x\d+<$/;
    if (!dimensionsPattern.test(dimensions)) {
      return reject(new Error(`Invalid dimensions format: "${dimensions}". Use formats like "800x600", "50%", "800x", "x600".`));
    }

    // Ensure output directory exists
    const outputDir = path.dirname(resolvedOutput);
    if (!fs.existsSync(outputDir)) {
      fs.mkdirSync(outputDir, { recursive: true });
    }

    // Ensure input file exists
    if (!fs.existsSync(resolvedInput)) {
      return reject(new Error(`Input file does not exist: ${resolvedInput}`));
    }

    // Build the ImageMagick convert command using an argument array to avoid shell injection
    const args = [
      'convert',
      resolvedInput,
      '-resize',
      dimensions,
      resolvedOutput,
    ];

    // Use execFile-style safety by joining with proper quoting via spawn,
    // but since exec is requested, we escape paths carefully.
    // For extra safety, use execFile from child_process instead:
    const { execFile } = require('child_process');

    execFile('convert', [resolvedInput, '-resize', dimensions, resolvedOutput], (error, stdout, stderr) => {
      if (error) {
        return reject(new Error(`ImageMagick error: ${stderr || error.message}`));
      }
      resolve(resolvedOutput);
    });
  });
}

// POST /upload - Upload and resize an image
app.post('/upload', upload.single('image'), async (req, res) => {
  try {
    if (!req.file) {
      return res.status(400).json({ error: 'No image file uploaded.' });
    }

    const { width, height, percentage } = req.body;

    // Determine resize dimensions
    let dimensions;
    if (percentage) {
      const pct = parseInt(percentage, 10);
      if (isNaN(pct) || pct <= 0 || pct > 1000) {
        return res.status(400).json({ error: 'Invalid percentage value. Must be between 1 and 1000.' });
      }
      dimensions = `${pct}%`;
    } else if (width && height) {
      const w = parseInt(width, 10);
      const h = parseInt(height, 10);
      if (isNaN(w) || isNaN(h) || w <= 0 || h <= 0 || w > 10000 || h > 10000) {
        return res.status(400).json({ error: 'Invalid width or height. Must be positive integers up to 10000.' });
      }
      dimensions = `${w}x${h}`;
    } else if (width) {
      const w = parseInt(width, 10);
      if (isNaN(w) || w <= 0 || w > 10000) {
        return res.status(400).json({ error: 'Invalid width. Must be a positive integer up to 10000.' });
      }
      dimensions = `${w}x`;
    } else if (height) {
      const h = parseInt(height, 10);
      if (isNaN(h) || h <= 0 || h > 10000) {
        return res.status(400).json({ error: 'Invalid height. Must be a positive integer up to 10000.' });
      }
      dimensions = `x${h}`;
    } else {
      dimensions = '800x600'; // Default dimensions
    }

    const inputFilePath = req.file.path;
    const ext = path.extname(req.file.filename);
    const baseName = path.basename(req.file.filename, ext);
    const outputFileName = `${baseName}_resized_${dimensions.replace('%', 'pct')}${ext}`;
    const outputFilePath = path.join(__dirname, 'resized', outputFileName);

    const resizedPath = await resizeImage(inputFilePath, outputFilePath, dimensions);

    res.status(200).json({
      message: 'Image resized successfully.',
      original: req.file.filename,
      resized: path.basename(resizedPath),
      dimensions,
      outputPath: resizedPath,
    });
  } catch (error) {
    console.error('Error resizing image:', error.message);
    res.status(500).json({ error: error.message });
  }
});

// GET /resized/:filename - Serve a resized image
app.get('/resized/:filename', (req, res) => {
  const { filename } = req.params;

  // Sanitize filename to prevent path traversal
  const safeFilename = path.basename(filename);
  const filePath = path.join(__dirname, 'resized', safeFilename);

  if (!fs.existsSync(filePath)) {
    return res.status(404).json({ error: 'File not found.' });
  }

  res.sendFile(filePath);
});

// Global error handler for multer errors
app.use((err, req

File: firstrun/sonnet/claude_sonnet/file_004_naive.py
import os
import subprocess
import shutil
from pathlib import Path
from flask import Flask, request, jsonify

app = Flask(__name__)

UPLOAD_FOLDER = "uploads"
RESIZED_FOLDER = "resized"

os.makedirs(UPLOAD_FOLDER, exist_ok=True)
os.makedirs(RESIZED_FOLDER, exist_ok=True)

ALLOWED_EXTENSIONS = {"png", "jpg", "jpeg", "gif", "bmp", "webp"}


def allowed_file(filename: str) -> bool:
    return "." in filename and filename.rsplit(".", 1)[1].lower() in ALLOWED_EXTENSIONS


def sanitize_filename(filename: str) -> str:
    """Remove any path components and keep only the base filename."""
    return os.path.basename(filename)


def resize_image(input_path: str, output_path: str, width: int, height: int) -> dict:
    """
    Resize an image using ImageMagick's 'convert' command via subprocess.

    Args:
        input_path: Absolute path to the input image file.
        output_path: Absolute path where the resized image will be saved.
        width: Target width in pixels.
        height: Target height in pixels.

    Returns:
        A dictionary with success status and a message.
    """
    # Validate that width and height are positive integers
    if not isinstance(width, int) or not isinstance(height, int):
        return {"success": False, "message": "Width and height must be integers."}
    if width <= 0 or height <= 0:
        return {"success": False, "message": "Width and height must be positive."}
    if width > 10000 or height > 10000:
        return {"success": False, "message": "Width and height must not exceed 10000."}

    # Resolve paths to prevent directory traversal
    input_path = str(Path(input_path).resolve())
    output_path = str(Path(output_path).resolve())

    # Ensure the input file exists
    if not os.path.isfile(input_path):
        return {"success": False, "message": f"Input file not found: {input_path}"}

    # Check that ImageMagick 'convert' is available
    if not shutil.which("convert"):
        return {
            "success": False,
            "message": "ImageMagick 'convert' command not found. Please install ImageMagick.",
        }

    geometry = f"{width}x{height}!"

    # Build the command as a list to avoid shell injection
    command = ["convert", input_path, "-resize", geometry, output_path]

    try:
        result = subprocess.run(
            command,
            capture_output=True,
            text=True,
            timeout=30,
            check=False,  # We handle returncode manually
        )

        if result.returncode != 0:
            return {
                "success": False,
                "message": f"ImageMagick error: {result.stderr.strip()}",
            }

        return {
            "success": True,
            "message": "Image resized successfully.",
            "output_path": output_path,
        }

    except subprocess.TimeoutExpired:
        return {"success": False, "message": "Image resize operation timed out."}
    except Exception as e:
        return {"success": False, "message": f"Unexpected error: {str(e)}"}


@app.route("/upload-and-resize", methods=["POST"])
def upload_and_resize():
    """
    Endpoint to upload an image and resize it.

    Expects:
        - A multipart/form-data request with:
            - 'image': The image file to upload.
            - 'width': Target width (integer).
            - 'height': Target height (integer).

    Returns:
        JSON response with success status and message.
    """
    if "image" not in request.files:
        return jsonify({"success": False, "message": "No image file provided."}), 400

    file = request.files["image"]

    if file.filename == "":
        return jsonify({"success": False, "message": "No file selected."}), 400

    if not allowed_file(file.filename):
        return (
            jsonify(
                {
                    "success": False,
                    "message": f"File type not allowed. Allowed types: {', '.join(ALLOWED_EXTENSIONS)}",
                }
            ),
            400,
        )

    try:
        width = int(request.form.get("width", 800))
        height = int(request.form.get("height", 600))
    except ValueError:
        return (
            jsonify({"success": False, "message": "Width and height must be integers."}),
            400,
        )

    safe_filename = sanitize_filename(file.filename)
    input_path = os.path.join(UPLOAD_FOLDER, safe_filename)
    output_path = os.path.join(RESIZED_FOLDER, f"resized_{safe_filename}")

    # Resolve absolute paths
    input_path = str(Path(input_path).resolve())
    output_path = str(Path(output_path).resolve())

    # Ensure paths are within the expected directories
    upload_folder_resolved = str(Path(UPLOAD_FOLDER).resolve())
    resized_folder_resolved = str(Path(RESIZED_FOLDER).resolve())

    if not input_path.startswith(upload_folder_resolved):
        return jsonify({"success": False, "message": "Invalid file path."}), 400

    if not output_path.startswith(resized_folder_resolved):
        return jsonify({"success": False, "message": "Invalid output path."}), 400

    # Save the uploaded file
    file.save(input_path)

    # Resize the image
    result = resize_image(input_path, output_path, width, height)

    if result["success"]:
        return jsonify(result), 200
    else:
        return jsonify(result), 500


@app.route("/health", methods=["GET"])
def health():
    return jsonify({"status": "ok"}), 200


if __name__ == "__main__":
    app.run(debug=False, host="0.0.0.0", port=5000)